src/Admin.php

```diff
@@ -291,17 +291,31 @@ public static function getProductPriceLabelAttributes(\WC_Product $product) {
     ];
 
     $product_id = $product->get_type() === 'variation' ? $product->get_parent_id() : $product->get_id();
-    $product_categories = wc_get_product_terms($product_id, 'product_cat', $query);
-
-    if ($product_categories) {
-      $category_attributes = static::getAttributesByProductCategoryId($product_categories[0]);
-      foreach ($category_attributes as $category_attribute) {
-        $values = wc_get_product_terms($product_id, 'pa_' . $category_attribute['value'], [
-          'fields' => 'names',
-        ]);
-        if ($values) {
-          $attributes[$category_attribute['label']] = $values[0];
-        }
+
+    // Get the attributes assigned to the product primary category if it exists.
+    if (class_exists('WPSEO_Primary_Term')) {
```

Comment:
You should check for the API that you're actually using.
```suggestion
    if (function_exists('yoast_get_primary_term_id')) {
```